packages/fx-core/src/plugins/resource/bot/utils/resourceNameFactory.ts

```diff
@@ -2,20 +2,13 @@
 // Licensed under the MIT license.
 
 import { Alias, RegularExprs } from "../constants";
-import { SomethingMissingError } from "../errors";
-import { CommonStrings } from "../resources/strings";
-import * as utils from "./common";
 
 export class ResourceNameFactory {
-    public static createCommonName(appName: string | undefined, limit?: number): string {
-        if (!appName) {
-            throw new SomethingMissingError(CommonStrings.SHORT_APP_NAME);
-        }
-
+    public static createCommonName(appName: string, suffix: string, limit?: number): string {
```

Comment:
I am ok with appName? to avoid use app.short!

Reply:
I'm ok if using appName? and exchange order of appName and suffix since required param should not follow optional one.